test/specs/wrapper/overview.spec.js

```diff
@@ -0,0 +1,336 @@
+        const wrapper = mountingMethod({ template: '<div class="test"></div>' })
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Not visible):',
+          '',
+          'Html:',
+          '    <div class="test"></div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {}',
+          ''
+        ]
+
+        wrapper.isVisible = () => false
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('html', () => {
+      it('prints Html as empty when html is not defined', () => {
+        const wrapper = mountingMethod({ template: '' })
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {}',
+          ''
+        ]
+
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('data', () => {
+      it('prints Data as {} when data is empty', () => {
+        const wrapper = mountingMethod({
+          template: '<div class="test" style="{ color: red }"></div>',
+          computed: {
+            onePlusOne: () => 1 + 1
+          }
+        })
+        wrapper.vm.$emit('foo', 'hello', 'world')
+        wrapper.vm.$emit('foo', 'bye', 'world')
+        wrapper.vm.$emit('bar', 'hey')
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '    <div class="test"></div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {',
+          '    onePlusOne: 2',
+          '}',
+          '',
+          'Emitted: {',
+          '    foo: [',
+          '        0: [ hello, world ],',
+          '        1: [ bye, world ]',
+          '    ],',
+          '    bar: [',
+          '        0: [ hey ]',
+          '    ]',
+          '}',
+          ''
+        ]
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('computed', () => {
+      it('prints Computed as {} when data is empty', () => {
+        const wrapper = mountingMethod({
+          template: '<div class="test" style="{ color: red }"></div>'
+        })
+        wrapper.vm.$emit('foo', 'hello', 'world')
+        wrapper.vm.$emit('foo', 'bye', 'world')
+        wrapper.vm.$emit('bar', 'hey')
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '    <div class="test"></div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {',
+          '    foo: [',
```

Comment:
what is displayed if objects are emitted? Probably need a test for this

Reply:
I totally agree. I'll get right on it.

Reply:
sounds good